common/pipeline_storage/src/main/scala/uk/ac/wellcome/pipeline_storage/PipelineStorageStream.scala

```diff
@@ -1,22 +1,22 @@
 package uk.ac.wellcome.pipeline_storage
 
 import akka.stream.FlowShape
-import akka.{Done, NotUsed}
 import akka.stream.scaladsl.{Broadcast, Flow, GraphDSL, Merge}
+import akka.{Done, NotUsed}
 import grizzled.slf4j.Logging
 import io.circe.Decoder
 import software.amazon.awssdk.services.sqs.model.Message
 import uk.ac.wellcome.messaging.MessageSender
 import uk.ac.wellcome.messaging.sqs.SQSStream
 
-import scala.concurrent.{ExecutionContext, Future}
-import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future, TimeoutException}
 
 case class PipelineStorageConfig(batchSize: Int,
                                  flushInterval: FiniteDuration,
                                  parallelism: Int)
 
-case class Bundle[T](message: Message, item: T)
+case class Bundle[T](message: Message, item: T, numberOfItems: Int)
```

Comment:
If T were type constrained to an Iterable, would you be able to drop `numberOfItems`?

Reply:
I don't think so...
the problem is a single message can result in many documents to ingest and send. We want to batch those independently to how many were related to the same original message, but we want to delete the message only when we know the ingest (and sending) was successful.

To ingest them independently of the original message we squash them together with documents reulting from other messages, but to delete the message when all of them have ingested successfully, we need to count how many we get at the end. That's what `numberOfItems` is used for. Before we squash them, we take note of how many documents per eacch message we have to ingest. After we've ingested, we count how many were successful. If the number match it's fine